Add qce paper to gh page

KuechA opened the pull request.

oxisto commented:
Should we add it to the readme as well? Or refer to the papers page in the readme?

KuechA commented:
> Should we add it to the readme as well? Or refer to the papers page in the readme?

I added a link to the "publications" site and also added the paper in the readme
